# In-class Activity #10 (August 4, 2026)

In [1]:
library(tidyverse)
library(moderndive)
library(car)

── Attaching core tidyverse packages ─────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ───────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Loading required package: carData


Attaching package: ‘car’


The following object is masked from ‘package:dplyr’:

    recode


The following object is masked from ‘package:purrr’:

    some




In [2]:
wage_data <- read.table("data/wage.txt", header=TRUE)
head(wage_data)

,education,south,sex,experience,union,wage,age,race,occupation,sector,marr
,<int>,<int>,<int>,<int>,<int>,<dbl>,<int>,<int>,<int>,<int>,<int>
1,8,0,1,21,0,5.10,35,2,6,1,1
2,9,0,1,42,0,4.95,57,3,6,1,1
3,12,0,0,1,0,6.67,19,3,6,1,0
4,12,0,0,4,0,4.00,22,3,6,0,0
5,12,0,0,17,0,7.50,35,3,6,0,1
6,13,0,0,9,1,13.07,28,3,6,0,0


### Part A

In [3]:
model_1 <- lm(wage ~ education, data = wage_data)
model_2 <- lm(wage ~ education + sex, data = wage_data)
summary(model_1)$adj.r.squared
summary(model_2)$adj.r.squared

[1] 0.1442589

[1] 0.1853039

When comparing the two models, model 2 appears to have a higher proportion of total variance of the response explained by the model while penalizing the addition of predictors (i.e. it exhibits a higher adjusted R^2 value) compared to model 1. More specifically, model 1 has an adjusted R^2 value of `0.144` whereas model 2 has an adjusted R^2 value of `0.185`. Given that we are examining the adjusted R^2 value and not the raw R^2 value, this increase is not simply attributed to adding additional variables to the model. Therefore, model 2 is a better model. Note, this does not tell us anything about the significance of adding the sex predictor to the model.

### Part B

In [4]:
anova(model_1, model_2)

,Res.Df,RSS,Df,Sum of Sq,F,Pr(>F)
,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
1,532,12023.41,NA,NA,NA,NA
2,531,11425.20,1,598.2092,27.8025,1.958444e-07


In the table above, model 1 is the reduced model whereas model 2 is the full model (which includes the additional predictor sex). The F-test compares the nested models by determining if the sex predictor is significant.

According to the table above, adding model 2 reduces the Sum of Squares by `598.2` which indicates that model 2 is explaining more of the variance in the data compared to model 1.

The p-value in the table above is `1.95e-07` which is significantly lower than the `0.05` significance level. Therefore at a `0.05` significance level, we reject the null hypothesis that the sex predictor has no effect on the model.

Therefore, model 2 is significantly better than model 1.